### PR TITLE
feat: tree view for project submission structure, cleanups

### LINKED ISSUE #382: create a tree view for structure checks
make user able to see edit and create structure checks from a tree view

### PR DESCRIPTION
**Pull request by EwoutV**
- Cleaned up the data fetching logic and loading states on all views
- Added an interactive project structure builder for automatic checks
- Improved the way in which forms worked: a single form component using a model
- Better form error displayment

closes #382

### CODE REVIEW

**Review by Topvennie — CHANGES_REQUESTED** [review]

File: backend/api/logic/parse_zip_files.py [hunk 1]
```diff
@@ -12,8 +12,8 @@ def parse_zip(project: Project, zip_file: InMemoryUploadedFile) -> bool:
 
     zip_file.seek(0)
 
-    with zipfile.ZipFile(zip_file, 'r') as zip:
-        files = zip.namelist()
+    with zipfile.ZipFile(zip_file, 'r') as file:
+        files = file.namelist()
```
[the remark this answers is not in the document]
Comment (EwoutV) [reply backend/api/logic/parse_zip_files.py:16]: zip was a shadowed build-in name noob

File: backend/api/views/project_view.py [hunk 2]
```diff
@@ -22,6 +25,9 @@
 from rest_framework.viewsets import GenericViewSet
 
 
+logger = logging.getLogger("ypovoli")
+
+
```
[the remark this answers is not in the document]
Comment (EwoutV) [reply backend/api/views/project_view.py:30]: yup 😿

File: backend/api/logic/parse_zip_files.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Topvennie) [reply backend/api/logic/parse_zip_files.py:16]: built-in names are there to be overwritten

**Review by DeLany123 — COMMENTED** [review]
Translations are not working properly
image

**Comment by DeLany123** [comment]
div moves when hovering "nieuwe map"
image

**Comment by DeLany123** [comment]
When pressing add new map, when the previous one is still named as "new" nothing happens, this is very unclear for a user. I suggest you change this to a placeholder which tells the user to fill in a name for a map so this almost never occurs.
image

File: frontend/src/views/authentication/LoginView.vue [hunk 3]
```diff
@@ -53,7 +53,7 @@ const { t } = useI18n();
     </BaseLayout>
 </template>
 
-<style lang="scss" scoped>
+<style lang="scss">
```
Comment (DeLany123) [inline frontend/src/views/authentication/LoginView.vue:56]: Please add scoped back (:

File: frontend/src/views/courses/roles/StudentCourseView.vue [hunk 4]
```diff
@@ -47,28 +43,28 @@ async function leaveCourse(): Promise<void> {
...
+        accept: async (): Promise<void> => {
             if (user.value !== null) {
                 // Leave the course
-                studentLeaveCourse(props.course.id, user.value.id).then(() => {
-                    // Refresh the user so the course is removed from the user's courses
-                    refreshUser();
-                    // Redirect to the dashboard
-                    push({ name: 'dashboard' });
-                });
+                await studentLeaveCourse(props.course.id, user.value.id);
```
Comment (DeLany123) [inline frontend/src/views/courses/roles/StudentCourseView.vue:49]: Shouldn't this be an emit too?

Comment (DeLany123) [reply frontend/src/views/courses/roles/StudentCourseView.vue:49]: I guess it does not matter because it gets redirected to the dashboard anyway

**Review by DeLany123 — COMMENTED** [review]
Monster pr, good job soldier!

File: backend/api/logic/parse_zip_files.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (EwoutV) [reply backend/api/logic/parse_zip_files.py:16]: yappa

File: frontend/src/views/courses/roles/StudentCourseView.vue (hunk 4 above)
Comment (EwoutV) [reply frontend/src/views/courses/roles/StudentCourseView.vue]: > Shouldn't this be an emit too?

It's a view, so no

**Pull request merged**